drivers/ms5607/MS5607.cpp

```diff
@@ -0,0 +1,392 @@
+
+#include <string.h>
+#include "DriverFramework.hpp"
+#include "MS5607.hpp"
+
+using namespace DriverFramework;
+
+#define ADDR_RESET_CMD				0x1E	/* write to this address to reset chip */
+#define ADDR_CMD_CONVERT_D1_OSR256		0x40	/* write to this address to start pressure conversion */
+#define ADDR_CMD_CONVERT_D1_OSR512		0x42	/* write to this address to start pressure conversion */
+#define ADDR_CMD_CONVERT_D1_OSR1024		0x44	/* write to this address to start pressure conversion */
+#define ADDR_CMD_CONVERT_D1_OSR2048		0x46	/* write to this address to start pressure conversion */
+#define ADDR_CMD_CONVERT_D1_OSR4096		0x48	/* write to this address to start pressure conversion */
+#define ADDR_CMD_CONVERT_D2_OSR256		0x50	/* write to this address to start temperature conversion */
+#define ADDR_CMD_CONVERT_D2_OSR512		0x52	/* write to this address to start temperature conversion */
+#define ADDR_CMD_CONVERT_D2_OSR1024		0x54	/* write to this address to start temperature conversion */
+#define ADDR_CMD_CONVERT_D2_OSR2048		0x56	/* write to this address to start temperature conversion */
+#define ADDR_CMD_CONVERT_D2_OSR4096		0x58	/* write to this address to start temperature conversion */
+#define ADDR_CMD_CONVERT_D1   ADDR_CMD_CONVERT_D1_OSR1024
+#define ADDR_CMD_CONVERT_D2   ADDR_CMD_CONVERT_D2_OSR1024
+
+#define ADDR_CMD_ADC_READ     0x00
+#define ADDR_PROM_SETUP       0xA0  /* address of 8x 2 bytes factory and calibration data */
+
+#define POW2(_x) ((_x) * (_x))
+
+// convertPressure must be called after convertTemperature
+// as convertTemperature sets m_raw_sensor_convertion values
+int64_t MS5607::convertPressure(int64_t adc_P)
+{
+	// Conversion from the datasheet
+	int64_t p = (((adc_P * m_raw_sensor_convertion.sens) >> 21) - m_raw_sensor_convertion.off) >> 15;
+	m_raw_sensor_convertion.pressure_mbar = p;
+	return p;
+}
+
+int32_t MS5607::convertTemperature(int32_t adc_T)
+{
+	// Conversion from the datasheet
+	/* temperature offset (in ADC units) */
+	int32_t dT = adc_T - ((int32_t)m_sensor_calibration.c5_reference_temp << 8);
+
+	int64_t sens = ((int64_t)m_sensor_calibration.c1_pressure_sens << 16)
+		       + (((int64_t)m_sensor_calibration.c3_temp_coeff_pres_sens * dT) >> 7);
+
+	int64_t off = ((int64_t)m_sensor_calibration.c2_pressure_offset << 17)
+		      + (((int64_t)m_sensor_calibration.c4_temp_coeff_pres_offset * dT) >> 6);
+
+	/* absolute temperature in centidegrees - note intermediate value is outside 32-bit range */
+	int32_t temp =  2000 + (int32_t)(((int64_t)dT * m_sensor_calibration.c6_temp_coeff_temp) >> 23);
+
+	/* temperature compensation */
+	if (temp < 2000) {
+		int32_t t2 = POW2(dT) >> 31;
+
+		int64_t f = POW2((int64_t)temp - 2000);
+		int64_t off2 = 61 * f >> 4;
+		int64_t sens2 = 2 * f;
+
+		if (temp < -1500) {
+			int64_t f2 = POW2(temp + 1500);
+			off2 += 15 * f2;
+			sens2 += 8 * f2;
+		}
+
+		temp -= t2;
+		sens -= sens2;
+		off -= off2;
+	}
+
+	m_raw_sensor_convertion.temperature_c = temp;
```

Comment:
If `temp` is in centidegrees, it should be `m_raw_sensor_convertion.temperature_c = temp / 100.0;`, right?